### PR TITLE
Handling missing translators

### PR DESCRIPTION
**Pull request by danielrischar**
Also alphabetizes error constants and adds a few more.

### CODE REVIEW

**Comment by arjun-msft** [comment]
:shipit:

File: node/lib/OpenT2TConstants.ts [hunk 1]
```diff
@@ -3,24 +3,21 @@
...
-    public static ResourceNotFound: string = "Resource not found";
     public static HMacSignatureVerificationFailed: string = "Payload signature doesn't match";
     public static InvalidAuthInfoInput: string =
-    "Invalid authInfo object. Please provide the existing authInfo object";
-
-    public static AccessDenied: string = "Access denied";
-    public static UnknownHubSubscribeRequest: string = "Unknown subscription request";
+        "Invalid authInfo object. Please provide the existing authInfo object";
+    public static InvalidHubId: string = "Invalid hub id";
+    public static InvalidResourceId: string = "Invalid reourceId";
```
Comment (shalinijoshi19) [inline node/lib/OpenT2TConstants.ts]: Could you fix the typo here -- "resourceId" (missing s) #Resolved

File: node/lib/OpenT2TConstants.ts [hunk 2]
```diff
@@ -3,24 +3,21 @@
...
+    public static MissingTranslator: string = "Cannot find translator module";
     public static MustSubscribeToDevice: string = "Must subscribe to a device";
-
+    public static NotImplemented: string = "Not Implemented";
+    public static ResourceNotFound: string = "Resource not found";
+    public static UnknownHubSubscribeRequest: string = "Unknown subscription request";
+    public static UnknownPlatform: string = "Unknown platform.  The requested platform cannot be translated";
```
Comment (shalinijoshi19) [inline node/lib/OpenT2TConstants.ts]: >  T [](start = 62, length = 3)

Nit: Double spaces?  #Resolved

File: node/lib/OpenT2T.ts [hunk 3]
```diff
@@ -96,6 +98,14 @@ export class OpenT2T {
 
         return new Promise<IThingTranslator>((resolve, reject) => {
             let translator: IThingTranslator;
+            
+            // Check for existance of the translator module
+            try {
+                require.resolve(translatorModuleName);
```
Comment (shalinijoshi19) [inline node/lib/OpenT2T.ts:104]: Again could you update the opent2t tests to check for this condition?  #Resolved

**Pull request merged**